src/main/java/space/chakat/verticality/mixin/MixinLivingEntity.java

```diff
@@ -22,19 +22,21 @@ private MixinLivingEntity() {
   }
 
   @Inject(
-      method = "canClimb",
-      at = @At(value = "RETURN", ordinal = 2),
-      locals = LocalCapture.CAPTURE_FAILHARD,
-      allow = 1,
-      cancellable = true)
-  void onCanClimb(
-      final CallbackInfoReturnable<Boolean> cir, final BlockState state, final Block block) {
+    method = "canClimb",
+    at = @At(value = "RETURN", ordinal = 2),
+    locals = LocalCapture.CAPTURE_FAILHARD,
+    allow = 1,
+    cancellable = true)
+  void onCanClimb(final CallbackInfoReturnable<Boolean> cir, final BlockState state, final Block block) {
     if (block instanceof Climbable) {
-      final LivingEntity thisEntity = (LivingEntity) (Object) this;
-      cir.setReturnValue(((Climbable) block).canClimb(thisEntity, state, this.getPos()));
-    } else if (world.getBlockState(new BlockPos(this.x, this.y - 0.5, this.z)).getBlock()
-        instanceof ClimbableFenceBlock) {
-      cir.setReturnValue(y - MathHelper.floor(y) < 0.5);
+      final LivingEntity self = (LivingEntity) (Object) this;
+      cir.setReturnValue(((Climbable) block).canClimb(self, state, getPos()));
+    } else {
+      final BlockPos down = new BlockPos(x, y - 0.5, z);
```

Comment:
Why breaking this into variables, and for that matter, the removal of `this`? Mostly just curious.

Reply:
You had an inconsistent use of `this.` qualifier (see fence return value), so I just removed the other qualifier usages, since they are redundant anyway ^^ I broke it up into variables for better readability but I can revert that if you want~